recipes/physx/4.x.x/conanfile.py

```diff
@@ -68,26 +58,27 @@ def config_options(self):
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
 
     def validate(self):
         if self.settings.os not in ["Windows", "Linux", "Macos", "Android", "iOS"]:
             raise ConanInvalidConfiguration("Current os is not supported")
 
-        if self.settings.os == "Macos" and self.settings.arch not in ["x86", "x86_64"]:
-            raise ConanInvalidConfiguration("{} only supports x86 and x86_64 on macOS".format(self.name))
+        if self.settings.os == "Macos":
+            if self.settings.arch not in ["x86", "x86_64"]:
+                raise ConanInvalidConfiguration("{} only supports x86 and x86_64 on macOS".format(self.name))
+            
+            if self.settings.os.version and Version(self.settings.os.version) < "10.14" and self.settings.compiler == "apple-clang" and valid_min_cppstd(self, 17):
```

Comment:
OS version is not set in c3i :)

Maybe?
```suggestion
            if (self.settings.os.version and Version(self.settings.os.version) < "10.14") or (self.settings.compiler == "apple-clang" and valid_min_cppstd(self, 17)):
```

I would just block `if self.setting.os == "Macos"` completely since a bunch of other configs fail, seems really complicated, the error could be

> macos is currently not supported by this recipe, contributions welcome

Reply:
I kinda did that, disabling MacOS compilation with clang and C++17 leaving room for contributions 🙂 
Hope it works 🤞  😅